pinot-controller/src/test/java/org/apache/pinot/controller/recommender/TestConfigEngine.java

```diff
@@ -331,23 +340,43 @@ void testFixedLenBitset() {
     //fixedLenBitset2 = {63, 64, 30, 25}};
     //fixedLenBitset3 = {2, 4, 9, 64, 77};
 
-    Assert.assertTrue(fixedLenBitset.contains(fixedLenBitset));
-    Assert.assertTrue(fixedLenBitset.contains(fixedLenBitset2));
-    Assert.assertFalse(fixedLenBitset.contains(fixedLenBitset3));
+    assertTrue(fixedLenBitset.contains(fixedLenBitset));
+    assertTrue(fixedLenBitset.contains(fixedLenBitset2));
+    assertFalse(fixedLenBitset.contains(fixedLenBitset3));
 
     for (int i = 0; i < 80; i++) {
       if (!fixedLenBitset.contains(i) && fixedLenBitset3.contains(i)) {
-        Assert.assertEquals(i, 77);
+        assertEquals(i, 77);
       }
     }
 
     fixedLenBitset.intersect(fixedLenBitset3);
     //fixedLenBitset = {2, 4, 9, 64};
 
-    Assert.assertEquals(fixedLenBitset.getCardinality(), 4);
-    Assert.assertTrue(fixedLenBitset.contains(2));
-    Assert.assertTrue(fixedLenBitset.contains(4));
-    Assert.assertTrue(fixedLenBitset.contains(64));
-    Assert.assertTrue(fixedLenBitset.contains(9));
+    assertEquals(fixedLenBitset.getCardinality(), 4);
+    assertTrue(fixedLenBitset.contains(2));
+    assertTrue(fixedLenBitset.contains(4));
+    assertTrue(fixedLenBitset.contains(64));
+    assertTrue(fixedLenBitset.contains(9));
+  }
+
+  @Test
+  void testRealtimeProvisioningRule() throws Exception {
+    String input = readInputToStr("recommenderInput/RealtimeProvisioningInput.json");
+    String output = RecommenderDriver.run(input);
+    ConfigManager configManager = objectMapper.readValue(output, ConfigManager.class);
+    Map<String, Map<String, String>> recommendations = configManager.getRealtimeProvisioningRecommendations();
+    assertRealtimeProvisioningRecommendation(recommendations.get(OPTIMAL_SEGMENT_SIZE));
+    assertRealtimeProvisioningRecommendation(recommendations.get(CONSUMING_MEMORY_PER_HOST));
+    assertRealtimeProvisioningRecommendation(recommendations.get(TOTAL_MEMORY_USED_PER_HOST));
+  }
+
+  private void assertRealtimeProvisioningRecommendation(Map<String, String> matrix) {
```

Comment:
This test is not very meaningful since it is not comparing/asserting against the actual output. It is essentially checking the format of the output. Why don't we compare the output agains the pre-buil 3 maps posted in the PR description ? Looks like that is indeed the output of the test.

Reply:
This test here makes sure that integration of recommendation engine with MemoryEstimator works by checking the size of the output based on the given parameters.